File: JS/day1.js
const fs = require('fs')

const input = require("../uploads/day1");

// part one
let answerA = 0;

for (let i = 0; i < input.length - 1; i++) {
  if (input[i] < input[i + 1]) answerA++;
}

console.log(answerA);

//part two
let answerB = 0;
for (let i = 0; i < input.length - 3; i++) {
  if(input[i] < input[i+3]) answerB++
}

console.log(answerB);

fs.writeFileSync('../out/day1.json', JSON.stringify({answerA, answerB}), {encoding:'utf-8', flag:'w'})

File: Python/day1.py

with open('uploads/day1.js', 'r') as file:
    filestring = file.read()


filestring = filestring.split('[  ')[1].split(',\n')[0:-1]


answerA = 0
for i in range(len(filestring)-1):
    if int(filestring[i]) < int(filestring[i+1]):
        answerA += 1

print(answerA)

answerB = 0
for i in range(len(filestring)-3):
    if int(filestring[i]) < int(filestring[i+3]):
        answerB = answerB+1

print(answerB)
